contributor-docs/adrs/adr-001-custom-elements.md

```diff
@@ -0,0 +1,115 @@
+
+### Extensibility
+Building behaviors in React Hooks gives us the ability to provide things like state and state change hooks to the consumer of the component. This allows the user to build on additional behaviors to the component based on the state or other variables provided to the component consumer. Doing the same with custom elements would require listening to events on the document[^3] and reacting to them. This is certainly do-able, but goes against some of the foundational principles of React (reacting to changes in the DOM vs changes in React state).
+
+### Organizational Overhead
+- GitHub’s custom elements are all managed in different repos which introduces more maintenance overhead.
+- Reacting to changes will take a bit more time, as we’ll need to wait for changes to be made to the custom elements until we can update Primer React Components (versus having behavior Hooks in PRC that we can update at our leisure).
+- Engineers who want to contribute to Primer React Components to build new components and behaviors would need to be familiar with both custom elements and React, two very different paradigms, and context switch between the two.
+
+### Other
+- The custom element & web component API progress slower than React due to changes needing to go through the whatwq standards process.
+
+### Risks of not switching to custom elements for behaviors
+- We spend extra time building behaviors in React that have already been built in our [custom elements library](https://github.github.io/web-systems-documentation/#custom-elements).
+  - There are currently 19 behaviors/components listed on the custom elements documentation site. Several of these we have already implemented in React in either PRC, Doctocat, or other React applications at GitHub which can be upstreamed (details-dialog, details-menu, clipboard-copy, text-expander, autocomplete, task-list via drag and drop hooks, tab-container, text-expander).
+- We decide not to invest further in React at GitHub and have wasted time we could have spent building more custom elements.
+  - This seems unlikely as there seems to be clear consensus that we will continue to build more and more highly interactive products.
+- The React library is abandoned and becomes obsolete
+  - This is a risk with any technology that we may use, seems highly unlikely in the near term, and is also a risk for custom elements.
+- Behaviors in .com using custom elements and behaviors in PRC diverge, leading to an inconsistent experience
+  - This is probably the biggest risk we face, but moving to custom elements isn’t necessarily the only or best solution. We should explore other ways of detecting divergence such as integration tests.
+
```

Comment:
In addition to these risks there's a large risk of duplicate efforts & engineering hours wasted developing interactions for two different stacks.

Reply:
This is addressed in the first bullet point